python/adaptivity.py.in

```diff
@@ -187,48 +194,64 @@ def patchwise_projection(dg_metric, mesh):
     # Implementation in C++ for speed
 
     code = r'''
-    #include <dolfin/mesh/Vertex.h>
-    #include <dolfin/fem/fem_utils.h>
+    #include <dolfin.h>
+    #include <pybind11/pybind11.h>
+    #include <pybind11/numpy.h>
+    #include <pybind11/pytypes.h>
+    #include <pybind11/stl.h>
+    #include <pybind11/cast.h>
+
+    namespace py = pybind11;
     namespace dolfin
     {
-      void project(const Function M, GenericVector& u, const Mesh mesh, const FunctionSpace space)
+      void project(const Function M, PETScVector& u, const Mesh mesh, std::shared_ptr<const FunctionSpace> space)
       {
-        const std::size_t dim = mesh.geometry().dim();
-        const std::vector<int> vertex_map = vertex_to_dof_map(space);
-        double init_vals[dim*dim];
-        memset( init_vals, 0, dim*dim*sizeof(double) );
-        Array<double> vals(dim*dim, init_vals);
-        double denom = 0.0;
-        double volume = 0.0;
-        unsigned count = 0;
-        unsigned len = u.local_size();
-        std::vector<double> values(len,0.0);
-        for (VertexIterator vertex(mesh); !vertex.end(); ++vertex)
-        {
-            for (CellIterator patch_cell(*vertex); !patch_cell.end(); ++patch_cell)
-            {
-                Array<double> p(dim,patch_cell->midpoint().coordinates());
-                M.eval(vals,p);
-                volume = patch_cell->volume();
-                for (unsigned i=0;i<dim*dim;i++)
-                {
-                values[vertex_map[count + i]] += vals[i]*volume;
-                }
-                denom += volume;
-            }
-            for (unsigned i=0;i<dim*dim;i++)
-            {
-            values[vertex_map[count + i]] = values[vertex_map[count + i]]/denom;
-            }
-            denom = 0.0;
-            count += dim*dim;
-        }
-        u.set_local(values);
+	const std::size_t dim = mesh.geometry().dim();
```

Comment:
Looks like indentation is not consistent here (either 2 spaces or nothing within { } )

Reply:
I will fix this